calyx-ir/src/control.rs

```diff
@@ -379,20 +555,65 @@ impl Cloner {
         }
     }
 
+    pub fn static_if(sif: &StaticIf) -> StaticIf {
+        StaticIf {
+            port: Rc::clone(&sif.port),
+            latency: sif.latency,
+            tbranch: Box::new(Self::static_(&sif.tbranch)),
+            fbranch: Box::new(Self::static_(&sif.fbranch)),
+            attributes: sif.attributes.clone(),
+        }
+    }
+
     pub fn par(par: &Par) -> Par {
         Par {
             stmts: par.stmts.iter().map(Self::control).collect(),
             attributes: par.attributes.clone(),
         }
     }
 
+    pub fn static_par(par: &StaticPar) -> StaticPar {
+        StaticPar {
+            stmts: par.stmts.iter().map(Self::static_).collect(),
+            attributes: par.attributes.clone(),
+            latency: par.latency,
+        }
+    }
+
     pub fn seq(seq: &Seq) -> Seq {
         Seq {
             stmts: seq.stmts.iter().map(Self::control).collect(),
             attributes: seq.attributes.clone(),
         }
     }
 
+    pub fn static_seq(seq: &StaticSeq) -> StaticSeq {
+        StaticSeq {
+            stmts: seq.stmts.iter().map(Self::static_).collect(),
+            attributes: seq.attributes.clone(),
+            latency: seq.latency,
+        }
+    }
+
+    pub fn static_(s: &StaticControl) -> StaticControl {
```

Comment:
complete nit: can we name this something else like `static_control`? I don't love the dangling underscore